datafusion/physical-plan/src/limit.rs

```diff
@@ -821,7 +816,7 @@ mod tests {
     #[tokio::test]
     async fn test_row_number_statistics_for_local_limit() -> Result<()> {
         let row_count = row_number_statistics_for_local_limit(4, 10).await?;
-        assert_eq!(row_count, Precision::Exact(10));
+        assert_eq!(row_count, Precision::Exact(40));
```

Comment:
This is interesting -- is the reason it changes because statistics represent data for all of a plans partitions and in this case there are 4 partitions, each with a limit of 10, and therefore the total output row count is 40?

Maybe as a follow on PR we could clarify that `Statistics` represent data for all partitions 🤔 

https://docs.rs/datafusion/latest/datafusion/common/struct.Statistics.html

Reply:
I have been thinking about this and I think it is supposed to be per-partition statistics. If we have a `with_fetch` method that takes in per-partition stats and a partition count, and it outputs global stats; we end up with a problem: We can not compose anymore.

I did a quick audit of the code and it seems to be used as per-partition stats everytwhere except the old `LocalLimitExec` `statistics` impl, which was IMHO erroneously converting to global stats -- which doesn't even make sense as it is a local operator.

I will send a commit with my proposed cleanup of this.